storages/jsonl-storage/src/lib.rs

```diff
@@ -0,0 +1,445 @@
+    async fn insert_schema(&mut self, schema: &Schema) -> Result<()> {
+        let data_path = self.data_path(schema.table_name.as_str());
+        File::create(data_path).map_storage_err()?;
+
+        if schema.column_defs.is_some() {
+            let schema_path = self.schema_path(schema.table_name.as_str());
+            let ddl = schema.to_ddl();
+            let mut file = File::create(schema_path).map_storage_err()?;
+            write!(file, "{ddl}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn delete_schema(&mut self, table_name: &str) -> Result<()> {
+        let data_path = self.data_path(table_name);
+        if data_path.exists() {
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist.to_string())?;
+        let table_path = JsonlStorage::data_path(self, table_name);
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(table_path)
+            .map_storage_err()?;
+
+        let column_defs = schema.column_defs.unwrap_or_default();
+        let labels = column_defs
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
+
+        for row in rows {
+            let json_string = match row {
+                DataRow::Vec(values) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in labels.iter().zip(values.into_iter()) {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+                DataRow::Map(hash_map) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in hash_map {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+            };
+            writeln!(file, "{json_string}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn insert_data(&mut self, table_name: &str, mut rows: Vec<(Key, DataRow)>) -> Result<()> {
+        let prev_rows = self.scan_data(table_name)?;
+        rows.sort_by(|(key_a, _), (key_b, _)| {
+            key_a
+                .partial_cmp(key_b)
+                .unwrap_or(std::cmp::Ordering::Equal)
+        });
+        let rows = rows.into_iter();
+
+        let sort_merge = SortMerge {
+            prev_rows,
+            rows,
+            current: None,
+        };
```

Comment:
How about adding `SortMerge::new`?
```rust
impl<T> SortMerge<T> where T: Iterator<Item = Result<(Key, DataRow)>> {
    fn new(prev_rows: T, rows: IntoIter<(Key, DataRow)>) -> Self {
        Self {
            prev_rows,
            rows,
            current: None,
        }
    }
}
```
Beause here, we have no use case of creating `SortMerge` using `current: Some(_)` so it would be better to hide `current` instance rather than exposing it.

Reply:
Then, this below would be enough.
```rust
let sort_merge = SortMerge::new(prev_rows, rows);
```